src/frame/headers.rs

```diff
@@ -635,7 +635,12 @@ impl HeaderBlock {
                     // Connection level header fields are not supported and must
                     // result in a protocol error.
 
-                    if name == header::CONNECTION {
+                    if name == header::CONNECTION
+                        || name == header::TRANSFER_ENCODING
+                        || name == header::UPGRADE
+                        || name == "keep-alive"
```

Comment:
It seems like this means it would be worth for `http` to provide consts for these headers, otherwise this will require iterating the string and lowercasing each char (of the RHS).

But this isn't an immediate requirement. I created hyperium/http/issues/142 to track.